### PR TITLE
feat(tianmu):DDL function filtering optimization is not supported(#1325)

### LINKED ISSUE #1325: bug: (Primary/Secondary)Tables that contain secondary indexes cannot be synchronized from the primary database to the secondary database.
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

```sql
###master
mysql> create table ttt(id int primary key,name varchar(10),key idx_name(name));
Query OK, 0 rows affected (0.01 sec)

mysql> show tables;
+--------------+
| Tables_in_db |
+--------------+
| ttt          |
+--------------+
1 row in set (0.00 sec)


###slave
mysql> show variables like 'sql_mode';
+---------------+----------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                        |
+---------------+----------------------------------------------------------------------------------------------+
| sql_mode      | STRICT_TRANS_TABLES,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION,MANDATORY_TIANMU,NO_KEY_ERROR |
+---------------+----------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> show tables;
Empty set (0.00 sec)
```

### Expected behavior

_No response_

### How To Reproduce

_No response_

### Environment

_No response_

### Are you interested in submitting a PR to solve the problem?

- [X] Yes, I will!

### PR DESCRIPTION
**Pull request by konghaiya**
Modify points:
1. Fix the error of judgment logic
2. Fix the value problem of (sql_mode) in the slave thread

## Summary about this PR

Issue Number: close #1325

## Tests Check List

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

## Changelog

- [ ] New Feature
- [x] Bug Fix
- [ ] Performance Improvement
- [ ] Build/Testing/CI/CD
- [ ] Documentation
- [ ] Not for changelog (changelog entry is not required)

## Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features

### CODE REVIEW

**Review by wisehead — APPROVED** [review]

File: sql/auth/sql_authorization.cc [hunk 1]
```diff
@@ -4389,9 +4389,11 @@ bool check_fk_parent_table_access(THD *thd,
 
   // Return if engine does not support Foreign key Constraint.
   if (!ha_check_storage_engine_flag(db_type, HTON_SUPPORTS_FOREIGN_KEYS)) {
+    sql_mode_t sql_mode = thd->variables.sql_mode;
+    if(thd->slave_thread) sql_mode = global_system_variables.sql_mode;
```
Comment (hustjieke) [inline sql/auth/sql_authorization.cc]: use a = if true ? ... : ..., as next:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```
Please rewrite this logical code in other files.

File: sql/sql_table.cc [hunk 2]
```diff
@@ -4080,15 +4082,19 @@ mysql_prepare_create_table(THD *thd, const char *error_schema_name,
                      MYF(0));
           DBUG_RETURN(TRUE);
         }
-        if ((create_info->db_type->db_type == DB_TYPE_TIANMU) &&
-            (!(thd->variables.sql_mode & MODE_NO_KEY_ERROR))) {
-          my_message(ER_TIANMU_NOT_SUPPORTED_FULLTEXT_INDEX,
-                     ER(ER_TIANMU_NOT_SUPPORTED_FULLTEXT_INDEX), MYF(0));
+        if (create_info->db_type->db_type == DB_TYPE_TIANMU){
+          sql_mode_t sql_mode = thd->variables.sql_mode;
```
Comment (hustjieke) [inline sql/sql_table.cc]: `sql_mode` assignment should be outside `if` condition.
Alse, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

File: sql/sql_table.cc [hunk 3]
```diff
@@ -8493,11 +8499,13 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
     // Now this contains only DROP for foreign keys and not-found objects
     Alter_drop *drop;
     drop_it.rewind();
+    sql_mode_t sql_mode = thd->variables.sql_mode;
```
Comment (hustjieke) [inline sql/sql_table.cc]: rewrite:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

File: sql/sql_table.cc [hunk 4]
```diff
@@ -8522,14 +8530,18 @@ mysql_prepare_alter_table(THD *thd, TABLE *table,
   }
   if (rename_key_list.elements)
   {
-    if ((create_info->db_type->db_type == DB_TYPE_TIANMU) &&
-        (!(thd->variables.sql_mode & MODE_NO_KEY_ERROR))) {
-      my_error(ER_TIANMU_NOT_FOUND_INDEX, MYF(0));
+    if (create_info->db_type->db_type == DB_TYPE_TIANMU){
+      sql_mode_t sql_mode = thd->variables.sql_mode;
```
Comment (hustjieke) [inline sql/sql_table.cc]: `sql_mode` assignment should be outside `if` condition.
Also, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

File: sql/sql_trigger.cc [hunk 5]
```diff
@@ -226,10 +226,13 @@ bool mysql_create_or_drop_trigger(THD *thd, TABLE_LIST *tables, bool create)
   }
   table= tables->table;
   table->pos_in_table_list= tables;
-  if ((table->file && table->file->ht == tianmu_hton) &&
-      (!(thd->variables.sql_mode & MODE_NO_KEY_ERROR))) {
-    my_error(ER_TIANMU_NOT_SUPPORTED_TRIGGER, MYF(0));
-    goto end;
+  if ((table->file && table->file->ht == tianmu_hton)){
+    sql_mode_t sql_mode = thd->variables.sql_mode;
```
Comment (hustjieke) [inline sql/sql_trigger.cc]: `sql_mode` assignment should be outside `if` condition.
Also, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

File: storage/tianmu/handler/ha_tianmu.cpp [hunk 6]
```diff
@@ -1683,11 +1683,14 @@ enum_alter_inplace_result ha_tianmu::check_if_supported_inplace_alter([[maybe_un
         ha_alter_info->handler_flags & Alter_inplace_info::DROP_PK_INDEX)
       DBUG_RETURN(HA_ALTER_INPLACE_NOT_SUPPORTED);
     // support alter table: mix add/drop key
+    sql_mode_t sql_mode = ha_thd()->variables.sql_mode;
```
Comment (hustjieke) [inline storage/tianmu/handler/ha_tianmu.cpp]: Also, rewrite code as follows:
```
sql_mode_t sql_mode = thd->slave_thread ? global_system_variables.sql_mode : thd->variables.sql_mode;
```

**Review by hustjieke — CHANGES_REQUESTED** [review]

File: sql/sql_table.cc (hunk 2 above)
Comment (konghaiya) [reply sql/sql_table.cc]: What does it mean outside the if condition?

**Review by duanfuxiang0 — APPROVED** [review]
LGTM

**Review by hustjieke — APPROVED** [review]
LGTM

**Pull request merged**